src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -108,16 +123,71 @@ private static RecoveryFlags GetSuppressionFlag(SyntaxBase precedingNode)
             }
         }
 
-        private SyntaxBase TargetScope()
+        private SyntaxBase TargetScope(IEnumerable<SyntaxBase> leadingNodes)
         {
             var keyword = ExpectKeyword(LanguageConstants.TargetScopeKeyword);
             var assignment = this.WithRecovery(this.Assignment, RecoveryFlags.None, TokenType.NewLine);
             var value = this.WithRecovery(() => this.Expression(allowComplexLiterals: true), RecoveryFlags.None, TokenType.NewLine);
 
-            return new TargetScopeSyntax(keyword, assignment, value);
+            return new TargetScopeSyntax(leadingNodes, keyword, assignment, value);
         }
 
-        private SyntaxBase ParameterDeclaration()
+        private SyntaxBase Decorator()
+        {
+            Token at = this.Expect(TokenType.At, b => b.ExpectedCharacter("@"));
+            SyntaxBase expression = this.WithRecovery(() =>
+            {
+                SyntaxBase current;
+                IdentifierSyntax identifier = this.Identifier(b => b.ExpectedNamespaceOrDecoratorName());
+
+                if (Check(TokenType.LeftParen))
+                {
+                    var functionCall = FunctionCallAccess(identifier, true);
+
+                    current = new FunctionCallSyntax(
+                        functionCall.Identifier,
+                        functionCall.OpenParen,
+                        functionCall.ArgumentNodes,
+                        functionCall.CloseParen);
+                }
+                else
+                {
+                    current = new VariableAccessSyntax(identifier);
+                }
+
+
+                while (this.Check(TokenType.Dot))

```

Comment:
Can you give examples of `PropertyAccessSyntax` and `InstanceFunctionCallSyntax` decorators?

Reply:
- `PropertyAccessSyntax`: `@az.secure`
- `InstanceFunctionCallSyntax`: `@az.minLength(1)`

Unfortunately `PropertyAccessSyntax` won't work for now, because currently the binder is not able to bind `secure` to a symbol.